From b2bd4a03f9c6e091e286536a6f1046c9d45875b9 Mon Sep 17 00:00:00 2001
From: Bharat Pawar <bpawar@codeaurora.org>
Date: Tue, 17 Oct 2017 15:32:23 +0530
Subject: [PATCH] ARM: dts: msm: Early mount of system partition for msm8916/39

Add support to early mount system partition so that system modules
can be loaded during early init.

Change-Id: I3fc377a2e8c2e8764a891d04031f416fd110cf55
Signed-off-by: Bharat Pawar <bpawar@codeaurora.org>
---
 arch/arm/boot/dts/qcom/msm8916.dtsi | 17 +++++++++++++++++
 arch/arm/boot/dts/qcom/msm8939.dtsi | 17 +++++++++++++++++
 2 files changed, 34 insertions(+)

diff --git a/arch/arm/boot/dts/qcom/msm8916.dtsi b/arch/arm/boot/dts/qcom/msm8916.dtsi
index 11b084b8c8c..62ae8421555 100644
--- a/arch/arm/boot/dts/qcom/msm8916.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8916.dtsi
@@ -28,6 +28,23 @@
 		bootargs = "sched_enable_hmp=1";
 	};
 
+        firmware: firmware {
+                 android {
+                         compatible = "android,firmware";
+                         fstab {
+                                 compatible = "android,fstab";
+                                 system {
+                                         compatible = "android,system";
+                                         dev = "/dev/block/platform/soc.0/7824900.sdhci/by-name/system";
+                                         type = "ext4";
+                                         mnt_flags = "ro,barrier=1,discard";
+                                         fsmgr_flags = "wait";
+                                         status = "ok";
+                                 };
+                         };
+                 };
+        };
+
 	aliases {
 		sdhc1 = &sdhc_1; /* SDC1 eMMC slot */
 		sdhc2 = &sdhc_2; /* SDC2 SD card slot */
diff --git a/arch/arm/boot/dts/qcom/msm8939.dtsi b/arch/arm/boot/dts/qcom/msm8939.dtsi
index f2e8bdb33a1..3b5688ee5d6 100644
--- a/arch/arm/boot/dts/qcom/msm8939.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8939.dtsi
@@ -29,6 +29,23 @@
 	chosen {
 		bootargs = "boot_cpus=0,4,5,6,7 sched_enable_hmp=1";
 	};
+
+        firmware: firmware {
+                 android {
+                         compatible = "android,firmware";
+                         fstab {
+                                 compatible = "android,fstab";
+                                 system {
+                                         compatible = "android,system";
+                                         dev = "/dev/block/platform/soc.0/7824900.sdhci/by-name/system";
+                                         type = "ext4";
+                                         mnt_flags = "ro,barrier=1,discard";
+                                         fsmgr_flags = "wait";
+                                         status = "ok";
+                                 };
+                         };
+                 };
+        };
 };
 
 &soc {
